java/src/main/java/com/powsybl/dataframe/network/extensions/ReferencePrioritiesDataframeProvider.java

```diff
@@ -0,0 +1,94 @@
+import com.powsybl.iidm.network.extensions.ReferencePriority;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+/**
+ * @author Damien Jeandemange <damien.jeandemange@artelys.com>
+ */
+@AutoService(NetworkExtensionDataframeProvider.class)
+public class ReferencePrioritiesDataframeProvider extends AbstractSingleDataframeNetworkExtension {
+
+    @Override
+    public String getExtensionName() {
+        return ReferencePriorities.NAME;
+    }
+
+    @Override
+    public ExtensionInformation getExtensionInformation() {
+        return new ExtensionInformation(ReferencePriorities.NAME,
+                "Defines the angle reference generator, busbar section or load of a power flow calculation, " +
+                        "i.e. which bus will be used with a zero-voltage angle.",
+                "index : id (str), " +
+                        "priority (int)");
+    }
+
+    private Stream<ReferencePriorities> itemsStream(Network network) {
+        return Stream.of(network.getGenerators(), network.getBusbarSections(), network.getLoads())
+                .flatMap(i -> StreamSupport.stream(i.spliterator(), false))
+                .map(g -> (ReferencePriorities) g.getExtension(ReferencePriorities.class))
+                .filter(Objects::nonNull);
+    }
+
+    private ReferencePriorities getOrThrow(Network network, String id) {
+        Identifiable<?> identifiable = getAndCheckInjection(network, id);
+        ReferencePriorities referencePriorities = identifiable.getExtension(ReferencePriorities.class);
+        if (referencePriorities == null) {
+            throw new PowsyblException("Injection '" + id + "' has no ReferencePriorities extension");
+        }
+        return referencePriorities;
+    }
+
+    @Override
+    public NetworkDataframeMapper createMapper() {
+        return NetworkDataframeMapperBuilder.ofStream(this::itemsStream, this::getOrThrow)
+                .stringsIndex("id", ext -> ((Identifiable<?>) ext.getExtendable()).getId())
+                .ints("priority",
+                        rp -> ((List<ReferencePriority>) rp.getReferencePriorities()).get(0).getPriority(),
+                        (rp, priority) -> ReferencePriority.set((Injection) (((List<ReferencePriority>) rp.getReferencePriorities()).get(0).getTerminal().getConnectable()), priority))
+                .build();
+    }
+
+    @Override
+    public void removeExtensions(Network network, List<String> ids) {
+        ids.stream().filter(Objects::nonNull)
+                .map(network::getIdentifiable)
+                .filter(Objects::nonNull)
+                .forEach(i -> i.removeExtension(ReferencePriorities.class));
+    }
+
+    @Override
+    public NetworkElementAdder createAdder() {
+        return new ReferencePrioritiesDataframeAdder();
+    }
+
+    static Injection getAndCheckInjection(Network network, String id) {
```

Comment:
To move to NetworkUtils where there is already such kind of utilities ?

Reply:
indeed, thanks

done